easybuild/easyblocks/m/matlab.py

```diff
@@ -133,18 +129,45 @@ def install_step(self):
         if LooseVersion(self.version) >= LooseVersion('2016b'):
             change_dir(self.builddir)
 
-        cmd = "%s %s -v -inputFile %s %s" % (self.cfg['preinstallopts'], src, self.configfile, self.cfg['installopts'])
-        (out, _) = run_cmd(cmd, log_all=True, simple=False)
-
-        # check installer output for known signs of trouble
-        patterns = [
-            "Error: You have entered an invalid File Installation Key",
-        ]
-        for pattern in patterns:
-            regex = re.compile(pattern, re.I)
-            if regex.search(out):
-                raise EasyBuildError("Found error pattern '%s' in output of installation command '%s': %s",
-                                     regex.pattern, cmd, out)
+        # MATLAB installer ignores TMPDIR
+        tmpdir = ''
+        if self.cfg['tmpdir']:
+            tmpd = self.cfg['tmpdir']
+        else:
+            tmpd = os.getenv('TMPDIR', '')
+        if tmpd:
+            tmpdir = '-tmpdir %s' % tmpd
```

Comment:
@akesandgren EasyBuild always sets up a unique temporary directory per installation, so there's no need to introduce a custom `tmpdir` easyconfig parameter for MATLAB, you can simplify this to:

```python
tmpdir = "-tmpdir %s" % tempfile.mkdtemp()

Reply:
Ah yes, I forgot about that when i originally made this. The problem was that I needed a much larger tmpdir to get it to build/install...